scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts

```diff
@@ -66,25 +72,15 @@ export function AuthoringTopbarDirective(
                 return TranslationService.checkAvailability(scope.item);
             };
 
-            scope.authoringActionsFromExtensions = [];
-
             scope.triggerActionFromExtension = (actionToTrigger: IArticleAction) => {
                 actionToTrigger.onTrigger();
             };
 
-            const authoringActionsFromExtensions = flatMap(
-                Object.values(extensions).map((ext) => ext.activationResult),
-                (activationResult) =>
-                    activationResult.contributions &&
-                    activationResult.contributions.authoringActions
-                        ? activationResult.contributions.authoringActions
-                        : [],
-            );
+            scope.$watch('item', () => {
+                setActionsFromExtensions();
+            }, true);
 
-            Promise.all(authoringActionsFromExtensions.map((getPromise) => getPromise(scope.item)))
-                .then((actions) => {
-                    scope.authoringActionsFromExtensions = flatMap(actions);
-                });
+            setActionsFromExtensions();
```

Comment:
you can avoid this call, `scope.$watch` will call it once on init. so there could be just:
```ts
scope.$watch('item', setActionsFromExtensions, true);`
```